ideascube/serveradmin/catalog.py

```diff
@@ -126,6 +129,10 @@ def remove(self, package):
         self._removed.append(package)
 
     def commit(self):
+        visible_packages = get_config('ideascube', 'displayed-package-ids')
+        [visible_packages.remove(p.id) for p in self._removed if p.id in visible_packages]
```

Comment:
This is not all the same (in spirit)

My version (and your "to long" version) loop over self._removed.
Your other versions loop over visible_packages.

But at the end, all versions finish by looping over both self._removed and visible_packages, so its more a conceptual difference that a true one. (But it can be in another context)

The [some(foo) for foo in foos if test(foo)] seems to become the standard way of writing (composition of) map/filter.

Even Guido suggests to remove map/filter/reduce from python3 (http://www.artima.com/weblogs/viewpost.jsp?thread=98196)
From now on, I tend towards this syntax all the time.
It's now a habit I have to just "drop" the "[]" in my mind in this context and read the line as a powerful map/filter syntax.

I would prefer have a standard way of looping over a iterator while dropping the element to write something like `loopover(visible_packages.remove(p.id) for p in self._removed if p.id in visible_packages)` and avoid create a list. But loopover is missing in python.

I think that `visible_packages = [p for p in visible_packages if p.id not in self._removed]` would be ok for every one (?)